examples/bovw/plot_bovw.py

```diff
@@ -0,0 +1,154 @@
+"""
+===================
+Bag of Visual Words
+===================
+
+An illustration of a Bag of Visual Words (BoVW) approach using textons for
+image recognition [1]_. The proposed solution is a naive and simplified
+solution using a limited number of patches and words and do not rely on any
+other well known computer vision features. It aims at illustrating the BoVW
+under a limited processing time rather than achieving high classification
+performance.
+
+References
+----------
+.. [1] Varma, Manik, and Andrew Zisserman. "A statistical approach to material
+       classification using image patch exemplars." IEEE transactions on
+       pattern analysis and machine intelligence 31.11 (2009): 2032-2047.
+
+"""
+
+# Author: Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# License: BSD
+
+import numpy as np
+from scipy.misc import imread
+
+from sklearn.feature_extraction.image import extract_patches_2d
+from sklearn.decomposition import PCA
+from sklearn.model_selection import StratifiedKFold
+from sklearn.cluster import MiniBatchKMeans
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import confusion_matrix
+from sklearn.externals.joblib import Parallel, delayed
+
+from tudarmstadt import fetch_tu_darmstadt
+
+
+print(__doc__)
+
+
+def proc_image(path_image, rng, patch_size=(9, 9), max_patches=10000):
+    """ Function to extract couple of patches from an image """
+
+    # Read the current image
+    im = imread(path_image)
+    # Extract patches
+    patch = extract_patches_2d(im, patch_size=patch_size,
+                               max_patches=max_patches, random_state=rng)
+    return patch.reshape((max_patches, np.prod(patch_size) * len(im.shape)))
+
+
+def image_extraction_projection(path_image, dict_PCA, rng,
+                                patch_size=(9, 9), max_patches=100000):
+    """ Function to extract a couple of patches from an image and apply PCA """
+
+    # Read the current image
+    im = imread(path_image)
+    # Extract patches
+    patch = extract_patches_2d(im, patch_size=patch_size,
+                               max_patches=max_patches, random_state=rng)
+
+    return dict_PCA.transform(patch.reshape((patch.shape[0],
+                                             np.prod(patch_size) *
+                                             len(im.shape))))
+
+# Define the parameters in use afterwards
+patch_size = (9, 9)
+max_patches = 100
+n_jobs = -1
```

Comment:
we can't really do that in an example, can we? We have j_jobs=3 and n_jobs=4 in some examples, though I'm not sure if that is a good idea.